secret-store-api/src/main/java/org/keycloak/secretstore/api/RequestRewriter.java

```diff
@@ -94,6 +94,13 @@ public HttpServerExchange rewrite(HttpServerExchange httpServerExchange) throws
 
         String authorization = new String(Base64.getDecoder().decode(authorizationHeaderParts[1]));
         String[] parts = authorization.split(":");
+
+        if (parts.length != 2) {
+            logger.authorizationValueInvalid(authorization);
+            // we don't return a 403 from here, as there might be something down the line that might accept this
```

Comment:
Yes, this runs before KC kicks in. So, KC will reject it, unless someone else put another filter which would "accept" this empty credential.